devicedefender/source/DeviceDefender.cpp

```diff
@@ -37,34 +39,37 @@ namespace Aws
             OnTaskCancelledHandler &&onCancelled,
             void *cancellationUserdata) noexcept
             : OnTaskCancelled(std::move(onCancelled)), cancellationUserdata(cancellationUserdata),
-              m_allocator(allocator), m_status(ReportTaskStatus::Ready),
-              m_taskConfig{mqttConnection.get()->GetUnderlyingConnection(),
-                           ByteCursorFromString(thingName),
-                           aws_event_loop_group_get_next_loop(eventLoopGroup.GetUnderlyingHandle()),
-                           reportFormat,
-                           aws_timestamp_convert(taskPeriodSeconds, AWS_TIMESTAMP_SECS, AWS_TIMESTAMP_NANOS, NULL),
-                           aws_timestamp_convert(
-                               networkConnectionSamplePeriodSeconds,
-                               AWS_TIMESTAMP_SECS,
-                               AWS_TIMESTAMP_NANOS,
-                               NULL),
-                           ReportTask::s_onDefenderV1TaskCancelled,
-                           this},
-              m_lastError(0)
+              m_allocator(allocator), m_status(ReportTaskStatus::Ready), m_taskConfig{nullptr}, m_owningTask{nullptr},
+              m_lastError(0), m_mqttConnection{mqttConnection}, m_eventLoopGroup(eventLoopGroup)
         {
+            (void)networkConnectionSamplePeriodSeconds;
+            struct aws_byte_cursor thingNameCursor = Crt::ByteCursorFromString(thingName);
+            m_lastError =
+                aws_iotdevice_defender_config_create(&m_taskConfig, allocator, &thingNameCursor, reportFormat);
+            if (AWS_OP_SUCCESS == m_lastError)
```

Comment:
trivial: it's a little weird to save the `AWS_OP_` result in the same variable that's supposed to hold an `AWS_ERROR_`. They're conceptually two different types, even if they both happen to be ints 

consider either
```
if (aws_iotdevice_defender_config_create(...) == AWS_OP_SUCCESS)
```
or
```
int result = aws_iotdevice_defender_config_create(...);
if (result == AWS_OP_SUCESS)
    ...
```
```